### PR TITLE
[SPARK-25007][R]Add array_intersect/array_except/array_union/shuffle to SparkR

### PR DESCRIPTION
**Pull request by huaxingao**
## What changes were proposed in this pull request?

Add the R version of array_intersect/array_except/array_union/shuffle

## How was this patch tested?
Add test in test_sparkSQL.R

### CODE REVIEW

**Comment by huaxingao** [comment]
@felixcheung @HyukjinKwon Sorry I couldn't figure out how to make the ```sequence``` work in the other PR. I will work on this one first.

File: R/pkg/R/generics.R [hunk 1]
```diff
@@ -799,10 +807,18 @@ setGeneric("array_sort", function(x) { standardGeneric("array_sort") })
...
+#' @name NULL
+setGeneric("array_union", function(x, y) { standardGeneric("array_union") })
+
 #' @rdname column_collection_functions
 #' @name NULL
 setGeneric("arrays_zip", function(x, ...) { standardGeneric("arrays_zip") })
 
+#' @rdname column_collection_functions
+#' @name NULL
+setGeneric("shuffle", function(x) { standardGeneric("shuffle") })
```
Comment (felixcheung) [inline R/pkg/R/generics.R]: this should go below - this part of the list should be sorted alphabetically

**Review by felixcheung — COMMENTED** [review]
thx, minor comment

File: R/pkg/R/generics.R (hunk 1 above)
Comment (huaxingao) [reply R/pkg/R/generics.R]: @felixcheung 
Changed. Thanks!

**Review by felixcheung — APPROVED** [review]

**Comment by felixcheung** [comment]
merged to master.

**Review by HyukjinKwon — COMMENTED** [review]
LGTM too

**Pull request closed without merging**